File: Classes/Residence.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace UniMate.Classes
{
    public class Residence
    {
        public int Id { get; set; }
        [Display(Name = "Dirección")]
        public string address { get; set; }
        [Display(Name = "Localidad")]
        public string locality { get; set; }
        [Display(Name = "Número de habitaciones")]
        public int roomsNumber { get; set; }
        [Display(Name = "Precio")]
        public float cost { get; set; }
        [Display(Name = "Disponibilidad")]
        public bool availability { get; set; }
        [Display(Name = "Fecha de disponibilidad")]
        public DateTime availabiltyDate { get; set; }
        [Display(Name = "Amoblado")]
        public bool furnished { get; set; }
        [Display(Name = "Ambiente de la locación")]
        public string mood { get; set; }
        [Display(Name = "Calificación")]
        public int rating { get; set; }
        [Display(Name = "Comentarios")]
        public List<string> comments { get; set; }
    }
}

File: frontend/src/Residence.js
import residenceStyles from './Residence.module.css';

const Residence = (props) => { 
    return (
        <div className={[residenceStyles.residence, 'p-4'].join(" ")}>
            <h5 className={residenceStyles.data}>Dirección: {props.address}</h5>
            {props.availability ?
                <div><h5 className={residenceStyles.data}>Disponibilidad: Sí</h5> <h5 className={residenceStyles.data}>Fecha de disponibilidad: {props.availabilityDate}</h5></div>
                : <h5 className={residenceStyles.data}>Disponibilidad: No</h5>
            }
            <h5 className={residenceStyles.data}>Costo: ${props.cost} COP</h5>
            <h5 className={residenceStyles.data}>Número de habitaciones: {props.roomsNumber}</h5>
            <h5 className={residenceStyles.rating}>
                {props.rating < 1 ? <span className={residenceStyles.star}>☆</span> : <span className={residenceStyles.fullStar}>★</span>}
                {props.rating < 2 ? <span className={residenceStyles.star}>☆</span> : <span className={residenceStyles.fullStar}>★</span>}
                {props.rating < 3 ? <span className={residenceStyles.star}>☆</span> : <span className={residenceStyles.fullStar}>★</span>}
                {props.rating < 4 ? <span className={residenceStyles.star}>☆</span> : <span className={residenceStyles.fullStar}>★</span>}
                {props.rating < 5 ? <span className={residenceStyles.star}>☆</span> : <span className={residenceStyles.fullStar}>★</span>}
            </h5>

        </div>
    )
};

export default Residence;